src/Module/Contacts.php

```diff
@@ -0,0 +1,1137 @@
+	{
+		$o = '';
+
+		if (!$update) {
+			// We need the editor here to be able to reshare an item.
+			if (local_user()) {
+				$x = [
+					'is_owner' => true,
+					'allow_location' => $a->user['allow_location'],
+					'default_location' => $a->user['default-location'],
+					'nickname' => $a->user['nickname'],
+					'lockstate' => (is_array($a->user) && (strlen($a->user['allow_cid']) || strlen($a->user['allow_gid']) || strlen($a->user['deny_cid']) || strlen($a->user['deny_gid'])) ? 'lock' : 'unlock'),
+					'acl' => ACL::getFullSelectorHTML($a->user, true),
+					'bang' => '',
+					'visitor' => 'block',
+					'profile_uid' => local_user(),
+				];
+				$o = status_editor($a, $x, 0, true);
+			}
+		}
+
+		$contact = DBA::selectFirst('contact', ['uid', 'url', 'id'], ['id' => $contact_id]);
+
+		if (!$update) {
+			$o .= self::contacts_tab($a, $contact, 1);
+		}
+
+		if (DBA::isResult($contact)) {
+			$a->page['aside'] = "";
+
+			$profiledata = Contact::getDetailsByURL($contact["url"]);
+
+			if (local_user()) {
+				if (in_array($profiledata["network"], [Protocol::ACTIVITYPUB, Protocol::DFRN, Protocol::DIASPORA, Protocol::OSTATUS])) {
+					$profiledata["remoteconnect"] = System::baseUrl()."/follow?url=".urlencode($profiledata["url"]);
+				}
+			}
+
+			Profile::load($a, "", 0, $profiledata, true);
+			$o .= Contact::getPostsFromUrl($contact["url"], true, $update);
+		}
+
+		return $o;
+	}
+
+	private static function contact_posts($a, $contact_id)
+	{
+		$contact = DBA::selectFirst('contact', ['uid', 'url', 'id'], ['id' => $contact_id]);
+
+		$o = self::contacts_tab($a, $contact, 2);
+
+		if (DBA::isResult($contact)) {
+			$a->page['aside'] = "";
+
+			$profiledata = Contact::getDetailsByURL($contact["url"]);
+
+			if (local_user()) {
+				if (in_array($profiledata["network"], [Protocol::ACTIVITYPUB, Protocol::DFRN, Protocol::DIASPORA, Protocol::OSTATUS])) {
+					$profiledata["remoteconnect"] = System::baseUrl()."/follow?url=".urlencode($profiledata["url"]);
+				}
+			}
+
+			Profile::load($a, "", 0, $profiledata, true);
+			$o .= Contact::getPostsFromUrl($contact["url"]);
+		}
+
+		return $o;
+	}
+
+	public static function _contact_detail_for_template(array $rr)
+	{
+		$dir_icon = '';
+		$alt_text = '';
+
+		switch ($rr['rel']) {
+			case Contact::FRIEND:
+				$dir_icon = 'images/lrarrow.gif';
+				$alt_text = L10n::t('Mutual Friendship');
+				break;
+
+			case Contact::FOLLOWER;
+				$dir_icon = 'images/larrow.gif';
+				$alt_text = L10n::t('is a fan of yours');
+				break;
+
+			case Contact::SHARING;
+				$dir_icon = 'images/rarrow.gif';
+				$alt_text = L10n::t('you are a fan of');
+				break;
+
+			default:
+				break;
+		}
+
+		$url = Contact::magicLink($rr['url']);
+
+		if (strpos($url, 'redir/') === 0) {
+			$sparkle = ' class="sparkle" ';
+		} else {
+			$sparkle = '';
+		}
+
+		if ($rr['self']) {
+			$dir_icon = 'images/larrow.gif';
+			$alt_text = L10n::t('This is you');
+			$url = $rr['url'];
```

Comment:
This is the problem of using arrays compared to objects, we have no way of being sure it actually is a well-formed Contact database record. Sure, it will most likely work, but without extensive documentation (which is currently lacking), this could go wrong.

Again, I would like to present Exhibit A: #5470